application/controllers/Drawers.php

```diff
@@ -405,17 +406,39 @@ public function addDrawer($shouldReturnJSON = false) {
 
 	}
 
-	public function setSortOrder($drawerId, $sortOrder) {
-		$accessLevel = $this->user_model->getAccessLevel("drawer",$this->doctrine->em->getReference("Entity\Drawer", $drawerId));
+	public function setSortOrder($drawerId, $sortOrder, $returnJSON = false) {
```

Comment:
I don't think we actually need the returnJson param here, since this endpoint returns JSON either way. It's just that the current JS isn't actually trapping for errors. So, you could safely remove this param and just return the render_json() path at all of the places you're using it - it won't do anything for the classic interface either way.

Reply:
Updated `setSortOrder` to always respond with json